### PR TITLE
Add XDMoD job for production OpenShift NERC

### PR DESCRIPTION
**Pull request by tzumainn**

### CODE REVIEW

**Comment by tzumainn** [comment]
@rob-baron should I test this out in the NERC infra environment first... ?

**Comment by rob-baron** [comment]
If it works in ocp-staging, then put in the pull request and we will deploy on the nerc.  Generally things have work on the nerc once they work in ocp-staging.

The database is about 10 times slower on the nerc than it is in ocp-staging, so shredding and ingestion take longer.

**Review by rob-baron — CHANGES_REQUESTED** [review]
I don't think we need the xdmod_src directory, unless there is a bug that you have fixed there.  There is a bug that I fixed, though this was fixed in the etl configuration (a database field size was the incorrect length).

File: k8s/kube-base/job-xdmod-openshift-prod.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: xdmod-openshift-prod-job
+spec:
+  schedule: "0 1 */1 * *"
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply k8s/kube-base/job-xdmod-openshift-prod.yaml:6]: Ah, good idea - updated!

File: k8s/kube-base/job-xdmod-openshift-prod.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          volumes:
+            - name: vol-xdmod-init
+              configMap:
+                items:
+                  - key: xdmod_init.json
+                    path: xdmod_init.json
+                name: cm-xdmod-init-json
+            - name: vol-xdmod-conf
+              persistentVolumeClaim:
+                claimName: pvc-xdmod-conf
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply k8s/kube-base/job-xdmod-openshift-prod.yaml]: Okay - I've removed it, and updated xdmod-openshift-reporting.sh to pull down the config files. I *think* that's all that's needed, but let me know if I'm misunderstanding something!

File: k8s/kube-base/job-xdmod-openshift-prod.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            - name: vol-xdmod-src
+              persistentVolumeClaim:
+                claimName: pvc-xdmod-src
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply k8s/kube-base/job-xdmod-openshift-prod.yaml]: Removed!

**Review by tzumainn — COMMENTED** [review]
Thanks for the review! Let me know if there are additional issues.

File: k8s/kube-base/job-xdmod-openshift-prod.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: xdmod-openshift-prod-job
+spec:
+  schedule: "0 1 */1 * *"
+  jobTemplate:
+    spec:
+      concurrencyPolicy: "Forbid"
```
Comment (rob-baron) [inline k8s/kube-base/job-xdmod-openshift-prod.yaml:9]: shouldn't 'concurrencyPolicy: "Forbid"' be at the same level as 'schedule: "0 1 ..."'?

**Review by rob-baron — CHANGES_REQUESTED** [review]
I think you have concurrencyPolicy placed incorrectly - can you provide the example that you are working from?

File: k8s/kube-base/job-xdmod-openshift-prod.yaml (hunk 4 above)
Comment (tzumainn) [reply k8s/kube-base/job-xdmod-openshift-prod.yaml]: Oh, whoops - you're right. Fixed!

**Review by rob-baron — APPROVED** [review]

**Comment by tzumainn** [comment]
@rob-baron thanks! does this need one more approval to be merged?

**Comment by tzumainn** [comment]
I've cleaned up the staging server so you can take a look at how the job operates there - let me know if there's anything else I should do!

**Comment by rob-baron** [comment]
@tzumainn  then why are there so many xdmod-openshift-staging-job jobs in container creating and none of them complete.

Note the xdmod-ingestor job is running on a daily basis and completes.

Now using project "xdmod" on server "https://api.ocp-staging.massopen.cloud:6443".
Rob@Roberts-Computer xdmod-pr % oc get pods  
NAME                                            READY   STATUS              RESTARTS   AGE
mariadb-f5497d5cf-dq8wn                         1/1     Running             0          78d
xdmod-ingestor-27829440--1-8bz65                0/1     Completed           0          2d15h
xdmod-ingestor-27830880--1-tnkmd                0/1     Completed           0          39h
xdmod-ingestor-27832320--1-52rz6                0/1     Completed           0          15h
xdmod-openshift-staging-job-27721500--1-lgrr4   0/1     ContainerCreating   0          21h
xdmod-openshift-staging-job-27722940--1-khvnc   0/1     ContainerCreating   0          21h
xdmod-openshift-staging-job-27724380--1-c2r4r   0/1     ContainerCreating   0          75d
xdmod-openshift-staging-job-27725820--1-5xgdd   0/1     ContainerCreating   0          74d
xdmod-openshift-staging-job-27727260--1-wmr7v   0/1     ContainerCreating   0          73d
xdmod-openshift-staging-job-27728700--1-j4wcp   0/1     ContainerCreating   0          72d
xdmod-openshift-staging-job-27730140--1-r6nfz   0/1     ContainerCreating   0          71d
xdmod-openshift-staging-job-27731580--1-c6nhl   0/1     ContainerCreating   0          70d
xdmod-openshift-staging-job-27733020--1-kcbcx   0/1     ContainerCreating   0          69d
xdmod-openshift-staging-job-27734460--1-bk62s   0/1     ContainerCreating   0          68d
xdmod-openshift-staging-job-27735900--1-hcfns   0/1     ContainerCreating   0          67d
xdmod-openshift-staging-job-27737340--1-868s6   0/1     ContainerCreating   0          66d
xdmod-openshift-staging-job-27738780--1-ckmjg   0/1     ContainerCreating   0          65d
xdmod-openshift-staging-job-27740220--1-ptp2c   0/1     ContainerCreating   0          64d
xdmod-openshift-staging-job-27741660--1-5rgj6   0/1     ContainerCreating   0          63d
xdmod-openshift-staging-job-27743100--1-2spp6   0/1     ContainerCreating   0          62d
xdmod-openshift-staging-job-27744540--1-swzm8   0/1     ContainerCreating   0          61d
xdmod-openshift-staging-job-27745980--1-xwb9c   0/1     ContainerCreating   0          60d
xdmod-openshift-staging-job-27747420--1-nkmhg   0/1     ContainerCreating   0          59d
xdmod-openshift-staging-job-27748860--1-btqwk   0/1     ContainerCreating   0          58d
xdmod-openshift-staging-job-27750300--1-bztzx   0/1     ContainerCreating   0          57d
xdmod-openshift-staging-job-27751740--1-t7pkm   0/1     ContainerCreating   0          56d
xdmod-openshift-staging-job-27753180--1-jmcf4   0/1     ContainerCreating   0          55d
xdmod-openshift-staging-job-27754620--1-v8brm   0/1     ContainerCreating   0          54d
xdmod-openshift-staging-job-27756060--1-nghzd   0/1     ContainerCreating   0          53d
xdmod-openshift-staging-job-27757500--1-zlsp6   0/1     ContainerCreating   0          52d
xdmod-openshift-staging-job-27758940--1-q25bh   0/1     ContainerCreating   0          51d
xdmod-openshift-staging-job-27760380--1-phgc4   0/1     ContainerCreating   0          50d
xdmod-openshift-staging-job-27761820--1-xcj8n   0/1     ContainerCreating   0          49d
xdmod-openshift-staging-job-27763260--1-l79z7   0/1     ContainerCreating   0          48d
xdmod-openshift-staging-job-27764700--1-4lqn8   0/1     ContainerCreating   0          47d
xdmod-openshift-staging-job-27766140--1-rvslg   0/1     ContainerCreating   0          46d
xdmod-openshift-staging-job-27767580--1-dpbqh   0/1     ContainerCreating   0          45d
xdmod-openshift-staging-job-27769020--1-qjrgx   0/1     ContainerCreating   0          44d
xdmod-openshift-staging-job-27770460--1-7kghj   0/1     ContainerCreating   0          43d
xdmod-openshift-staging-job-27771900--1-cs8tj   0/1     ContainerCreating   0          42d
xdmod-openshift-staging-job-27773340--1-qwmbc   0/1     ContainerCreating   0          41d
xdmod-openshift-staging-job-27774780--1-b7klv   0/1     ContainerCreating   0          40d
xdmod-openshift-staging-job-27776220--1-2nh92   0/1     ContainerCreating   0          39d
xdmod-openshift-staging-job-27777660--1-wstqn   0/1     ContainerCreating   0          38d
xdmod-openshift-staging-job-27779100--1-stgqs   0/1     ContainerCreating   0          37d
xdmod-openshift-staging-job-27780540--1-fj689   0/1     ContainerCreating   0          36d
xdmod-openshift-staging-job-27781980--1-gkfk9   0/1     ContainerCreating   0          35d
xdmod-openshift-staging-job-27783420--1-jc7sb   0/1     ContainerCreating   0          34d
xdmod-openshift-staging-job-27784860--1-sqprs   0/1     ContainerCreating   0          33d
xdmod-openshift-staging-job-27786300--1-tnhbf   0/1     ContainerCreating   0          32d
xdmod-openshift-staging-job-27787740--1-4lstj   0/1     ContainerCreating   0          31d
xdmod-openshift-staging-job-27789180--1-p772s   0/1     ContainerCreating   0          30d
xdmod-openshift-staging-job-27790620--1-7kwrk   0/1     ContainerCreating   0          29d
xdmod-openshift-staging-job-27792060--1-8947q   0/1     ContainerCreating   0          28d
xdmod-openshift-staging-job-27793500--1-qsk4z   0/1     ContainerCreating   0          27d
xdmod-openshift-staging-job-27794940--1-55x8c   0/1     ContainerCreating   0          26d
xdmod-openshift-staging-job-27796380--1-vwgtj   0/1     ContainerCreating   0          25d
xdmod-openshift-staging-job-27797820--1-b764v   0/1     ContainerCreating   0          24d
xdmod-openshift-staging-job-27799260--1-gjgkd   0/1     ContainerCreating   0          23d
xdmod-openshift-staging-job-27800700--1-ccrl7   0/1     ContainerCreating   0          22d
xdmod-openshift-staging-job-27802140--1-mht7r   0/1     ContainerCreating   0          21d
xdmod-openshift-staging-job-27803580--1-bjx5g   0/1     ContainerCreating   0          20d
xdmod-openshift-staging-job-27805020--1-4knjc   0/1     ContainerCreating   0          19d
xdmod-openshift-staging-job-27806460--1-hhw7l   0/1     ContainerCreating   0          18d
xdmod-openshift-staging-job-27807900--1-xc76q   0/1     ContainerCreating   0          17d
xdmod-openshift-staging-job-27809340--1-k7b4f   0/1     ContainerCreating   0          16d
xdmod-openshift-staging-job-27810780--1-4qj57   0/1     ContainerCreating   0          15d
xdmod-openshift-staging-job-27812220--1-bd4tp   0/1     ContainerCreating   0          14d
xdmod-openshift-staging-job-27813660--1-sgzvg   0/1     ContainerCreating   0          13d
xdmod-openshift-staging-job-27815100--1-w2qgr   0/1     ContainerCreating   0          12d
xdmod-openshift-staging-job-27816540--1-lnpqq   0/1     ContainerCreating   0          11d
xdmod-openshift-staging-job-27817980--1-kpbqt   0/1     ContainerCreating   0          10d
xdmod-openshift-staging-job-27819420--1-cg2hg   0/1     ContainerCreating   0          9d
xdmod-openshift-staging-job-27820860--1-g9q6j   0/1     ContainerCreating   0          8d
xdmod-openshift-staging-job-27822300--1-hbc6v   0/1     ContainerCreating   0          7d14h
xdmod-openshift-staging-job-27823740--1-vst4b   0/1     ContainerCreating   0          6d14h
xdmod-openshift-staging-job-27825180--1-kkpcw   0/1     ContainerCreating   0          5d14h
xdmod-openshift-staging-job-27826620--1-d8rlt   0/1     ContainerCreating   0          4d14h
xdmod-openshift-staging-job-27828060--1-h27p4   0/1     ContainerCreating   0          3d14h
xdmod-openshift-staging-job-27829500--1-zz695   0/1     ContainerCreating   0          2d14h
xdmod-openshift-staging-job-27830940--1-jcnr8   0/1     ContainerCreating   0          38h
xdmod-openshift-staging-job-27832380--1-5rcvh   0/1     ContainerCreating   0          14h
xdmod-ui-586f5f88d-2bx9m                        1/1     Running             0          23d

**Comment by tzumainn** [comment]
Ah! I've been testing in the 'xdmod-tzumainn' project on staging. I didn't realize stuff had gotten installed in the 'xdmod' project.

**Comment by tzumainn** [comment]
I've just removed the cronjob from the 'xdmod' project to remove confusion there - my fault! You can see the job working in 'xdmod-tzumainn'.

**Pull request merged**